src/pause/138CC0.c

```diff
@@ -103,19 +103,386 @@ s32 pause_badges_scroll_offset_x(s32 x) {
+                        selectedRow = 0;
+                    }
+                    if (selectedRow < (s32) gBadgeMenuPages[gBadgeMenuCurrentPage].listStart) {
+                        gBadgeMenuCurrentPage -= 1;
+                    }
+                } else {
+                    // Z button press
+                    gBadgeMenuCurrentPage -= 1;
+                    if (gBadgeMenuCurrentPage < 0) {
+                        gBadgeMenuCurrentPage = 0;
+                    }
+                    selectedRow = (s32) gBadgeMenuPages[gBadgeMenuCurrentPage].listStart;
+                }
+            }
+
+            // Need to re-read button state here for strange regalloc reasons
+            heldButtons2 = gPauseMenuHeldButtons;
+            if (heldButtons2 & (BUTTON_STICK_DOWN | BUTTON_R)) {
+                if (heldButtons2 & BUTTON_STICK_DOWN) {
+                    PauseItemPage* page = &gBadgeMenuPages[gBadgeMenuCurrentPage];
+                    selectedRow += 1;
+                    if (selectedRow >= (page->listStart + page->numRows)) {
+                        gBadgeMenuCurrentPage += 1;
+                        if (gBadgeMenuPages[gBadgeMenuCurrentPage].enabled == 0) {
+                            gBadgeMenuCurrentPage -= 1;
+                            selectedRow -= 1;
+                        }
+                    }
+                } else {
+                    // R button press
+                    PauseItemPage* newPage = &gBadgeMenuPages[++gBadgeMenuCurrentPage];
+
+                    if (newPage->enabled == 0) {
+                        gBadgeMenuCurrentPage -= 1;
+                    } else {
+                        selectedRow = (s32) newPage->listStart;
+                    }
+                }
+            }
+
+            newPageNumCols = gBadgeMenuPages[gBadgeMenuCurrentPage].numCols;
+            if (gBadgeMenuItemIDs[selectedRow * newPageNumCols] != 0x7FFE) {
+                if (gPauseMenuHeldButtons & BUTTON_STICK_LEFT) {
+                    selectedCol -= 1;
+                    if (selectedCol < 0) {
+                        selectedCol = newPageNumCols - 1;
+                    }
+                } else if (gPauseMenuHeldButtons & BUTTON_STICK_RIGHT) {
+                    selectedCol += 1;
+                    if (selectedCol >= newPageNumCols) {
+                        selectedCol = 0;
+                    }
+                }
+            } else {
+                selectedCol = 0;
+            }
+            gBadgeMenuSelectedIndex = selectedCol + (selectedRow * gBadgeMenuPages[gBadgeMenuCurrentPage].numCols);
+            if (gBadgeMenuSelectedIndex != selectedIndex) {
+                sfx_play_sound(SOUND_MENU_CHANGE_SELECTION);
+            }
+            gBadgeMenuSelectedItemID = gBadgeMenuItemIDs[gBadgeMenuSelectedIndex];
+        }
+    }
+
+    if (gBadgeMenuLevel == 1) {
+        s32 itemID = gBadgeMenuSelectedItemID;
+        if (((itemID != 0x7FFE) && (itemID != 0x7FFF) && (itemID != 0))) {
```

Comment:
we should make defines for these 0x7FFE and 0x7FFF vals since they're everywhere...not sure what they are though. @nanaian any thoughts?

Reply:
0x7FFE and 0x7FFF are hinted at more by the draw function (not included here). `0x7FFE` is a fake item ID used for the "None" line if you select "Equipped Badges" but have none equipped. 0x7FFF is empty slot, looks like checks for it are mostly defensive coding.

I can add macros for `BADGE_NONE` and `BADGE_INVALID`